### PR TITLE
Add Cubex

### PR DESCRIPTION
**Pull request by MrEssex**
https://packagist.org/packages/cubex/framework

Something we use at work, would be nice to see where it ranks along the others

### CODE REVIEW

**Comment by waghanza** [comment]
Hi @bajb @TomK,

Would you mind to add (or consent that we add) [cubex](https://github.com/cubex/framework) in `here` ?

------------
This project compare performance based on 3 routes :

+ `GET` `/`, **SHOULD** return a successful status code and an empty string
+ `GET` `/user/:id`, **SHOULD** return a successful status code and the id
+ `POST` `/user`, **SHOULD** return a successful status code and an empty string

Regards,

**Comment by bajb** [comment]
Happy for that to be added.  The methods on there look correct

**Comment by TomK** [comment]
@waghanza works for me 👍

File: php/cubex/config.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+framework:
+  website: cubex.io
+  version: 4.15.0
```
Comment (waghanza) [inline php/cubex/config.yaml:3]: We prefer this format

```suggestion
  version: 4.15
```

File: php/cubex/composer.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "license": "MIT",
+  "type": "project",
+  "config": {
+    "optimize-autoloader": true,
+    "preferred-install": "dist",
+    "sort-packages": true
+  },
+  "prefer-stable": true,
+  "require": {
+    "cubex/framework": "^4.0"
```
Comment (waghanza) [inline php/cubex/composer.json:12]: better use `~` to be sure we are aligned with the doc
```suggestion
    "cubex/framework": "~4.15.0"
```

File: php/cubex/composer.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  "require-dev": {
```
Comment (waghanza) [inline php/cubex/composer.json:14]: not really necessary

File: php/cubex/composer.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "bajb/cubex-min",
+  "license": "MIT",
+  "type": "project",
+  "config": {
+    "optimize-autoloader": true,
+    "preferred-install": "dist",
+    "sort-packages": true
```
Comment (waghanza) [inline php/cubex/composer.json:8]: Better to remove this (those steps are explicitly made on _bootstrap_ step)

https://github.com/the-benchmarker/web-frameworks/blob/1ee1a7b896f7c1c6ddb2cf6d694e0337b30e078a/php/Dockerfile#L28-L30

**Review by waghanza — CHANGES_REQUESTED** [review]

**Comment by MrEssex** [comment]
Good morning - @waghanza 

Requested changes should now be committed

File: php/cubex/src/Application.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  protected function _generateRoutes()
+  {
+    yield self::_route('/user/{id}', 'user');
+    yield self::_route('/user', 'user');
+    yield self::_route('/', 'home');
+  }
+
+  public function postUser()
+  {
+    return "OK";
```
Comment (waghanza) [inline php/cubex/src/Application.php:17]: weird that CI does not complain, should be an empty string here

File: php/cubex/src/Application.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  public function getUser()
+  {
+    return $this->routeData()->get('id');
+  }
+
+  public function getHome()
+  {
+    return "OK";
```
Comment (waghanza) [inline php/cubex/src/Application.php:27]: weird that CI does not complain, should be an empty string here

**Review by waghanza — CHANGES_REQUESTED** [review]

File: php/cubex/src/Application.php (hunk 6 above)
Comment (MrEssex) [reply php/cubex/src/Application.php]: Fixed both of the above

**Review by waghanza — CHANGES_REQUESTED** [review]
Seems that CI has detected a failure on GET /user/0
https://github.com/the-benchmarker/web-frameworks/pull/3914/checks?check_run_id=1844945293#step:7:11

Thanks, btw, to let me notice that

**Comment by waghanza** [comment]
just a question @MrEssex @babj @TomK, does `cubex` support `PHP` **8** ?. we have an incoming `PR` https://github.com/the-benchmarker/web-frameworks/pull/3655

**Comment by MrEssex** [comment]
> Seems that CI has detected a failure on GET /user/0
> https://github.com/the-benchmarker/web-frameworks/pull/3914/checks?check_run_id=1844945293#step:7:11
> 
> Thanks, btw, to let me notice that

Apologies about that, odd thats its now failing. Will look into the reason why

**Review by waghanza — APPROVED** [review]
Thanks for your contribution 🎉

**Comment by MrEssex** [comment]
> just a question @MrEssex @Babj @TomK, does `cubex` support `PHP` **8** ?. we have an incoming `PR` #3655

It does out the box I believe, ran a little test not so long ago to ensure

**Pull request merged**